runners/core-java/src/main/java/org/apache/beam/runners/core/TimerInternals.java

```diff
@@ -236,6 +295,7 @@ public void encode(TimerData timer, OutputStream outStream) throws CoderExceptio
       STRING_CODER.encode(timer.getTimerId(), outStream);
       STRING_CODER.encode(timer.getNamespace().stringKey(), outStream);
       INSTANT_CODER.encode(timer.getTimestamp(), outStream);
+      INSTANT_CODER.encode(timer.getOutputTimestamp(), outStream);
```

Comment:
@kennknowles do you know how this coder is used? Does this change break update?

Reply:
I don't know offhand. I would look through the code. If it is used by `WindmillTimerInternals` then it would break update. We would need to make it able to decode the old format. That would only be forward update safe, not rollback safe.

Reply:
https://github.com/apache/beam/pull/9677/files#diff-162b02136a675bf8c25424d2d6fc377fR308

Are you talking about this?

Reply:
This change will cause incompatibility in the Dataflow runner. @kennknowles @dpmills do either of you have an idea of how to address this?  

One idea: in StreamingSideInputFetcher we could create a new state tag. Keep the old coder around for the old state tag, and fetch timers from both tags.